src/main/java/tools/map/making/ConnectionFinder.java

```diff
@@ -344,6 +345,8 @@ private static Point2D getCentroid(final Shape currentShape) {
             pointList.add(pointList.get(lastMoveToIndex));
           }
           break;
+        default:
+          throw new AssertionError("unknown path iterator segment type: " + segmentType);
```

Comment:
All current `PathIterator.SEG_*` values are handled, so the default case will only be triggered if the `PathIterator#currentSegment()` API is changed to return a new value without adding a corresponding case.